core/src/scene/styleParam.cpp

```diff
@@ -218,36 +194,42 @@ StyleParam::Value StyleParam::parseString(StyleParamKey key, const std::string&
     case StyleParamKey::text_transition_show_time:
     case StyleParamKey::text_transition_selected_time: {
         float time = 0.0f;
-        if (!parseTime(_value, time)) {
-            LOGW("Invalid time param '%s'", _value.c_str());
+        if (!node.IsScalar() || !parseTime(node.Scalar(), time)) {
+            LOGW("Invalid time param '%s'", Dump(node).c_str());
         }
         return time;
     }
     case StyleParamKey::text_font_family:
     case StyleParamKey::text_font_weight:
     case StyleParamKey::text_font_style: {
-        std::string normalized = _value;
-        std::transform(normalized.begin(), normalized.end(), normalized.begin(), ::tolower);
-        return normalized;
+        std::string result;
+        if (node.IsScalar()) {
+            result = node.Scalar();
+            std::transform(result.begin(), result.end(), result.begin(), ::tolower);
+        } else {
+            LOGW("Invalid font property: %s", Dump(node).c_str());
+        }
+        return result;
     }
     case StyleParamKey::anchor:
     case StyleParamKey::text_anchor: {
-        LabelProperty::Anchors anchors;
-        for (auto& anchor : splitString(_value, ',')) {
-            if (anchors.count == LabelProperty::max_anchors) { break; }
-
-            LabelProperty::Anchor labelAnchor;
-            if (LabelProperty::anchor(anchor, labelAnchor)) {
-                anchors.anchor[anchors.count++] = labelAnchor;
-            } else {
-                LOG("Invalid anchor %s", anchor.c_str());
+        LabelProperty::Anchors anchorSet;
+        LabelProperty::Anchor anchor;
+        if (node.IsScalar() && LabelProperty::anchor(node.Scalar(), anchor)) {
+            anchorSet.anchor[anchorSet.count++] = anchor;
+        } else if (node.IsSequence()) {
+            for (const auto& subNode : node) {
+                if (anchorSet.count >= LabelProperty::max_anchors) { break; }
```

Comment:
`anchorSet.count` can never become > `LabelProperty::max_anchors`. Equality check should suffice here.

Reply:
Just being paranoid, I don't think there's a downside to using `>=`.